pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -393,17 +393,9 @@ func (r *RepoSyncReconciler) handleReconcileError(ctx context.Context, err error
 			logFieldObjectRef, statusErr.ID.ObjectKey.String(),
 			logFieldObjectKind, statusErr.ID.Kind,
 			logFieldObjectStatus, statusErr.Status)
-		switch statusErr.Status {
-		case kstatus.InProgressStatus, kstatus.TerminatingStatus:
```

Comment:
@karlkfi PTAL, as a I may be missing something here.

It seems to be the case, at least for finalizer teardown, that we should not swallow this error - as that results in the finalizer exiting prematurely even if not all managed objects have been deleted.

Reply:
Returning the error causes the controller-runtime to immediately retry the Reconcile (with backoff). That's why it was returning nil. But there's nothing to do, so retrying immediately won't change anything.

Once the status is Stalled, anything waiting for completion should just keep waiting until the next even that removes the Stalled condition, right?

Reply:
We synced offline - TL;DR we decided to retain the Reconciling condition but return a `NoRetryError` so that the finalizer can be retained whilst also not immediately retriggering a Reconcile